pkgs/top-level/all-packages.nix

```diff
@@ -2732,6 +2732,8 @@ with pkgs;
 
   cron = isc-cron;
 
+  cudaLib = import ../development/cuda-modules/lib;
+
   cudaPackages_11_0 = callPackage ./cuda-packages.nix { cudaMajorMinorVersion = "11.0"; };
   cudaPackages_11_1 = callPackage ./cuda-packages.nix { cudaMajorMinorVersion = "11.1"; };
```

Comment:
Here - instead of exposing `cudaLib` I see something like `makeCudaScope = let cudaLib = ...; in cudaLib.${...}; cudaPackages_XX_YY = makeCudaScope { ... }`

Reply:
I would like to have `cudaLib` exposed outside of the CUDA package set since it does not depend on the CUDA version or architecture. Having it at the top level is also important because if out-of-tree consumers modify it (e.g., by adding new GPUs), those changes are propagated to all consumers of `cudaLib`. One such example where that's important is the `pkgsCuda` attribute set I'd like to introduce in https://github.com/NixOS/nixpkgs/pull/406568: https://github.com/NixOS/nixpkgs/pull/406568/commits/eb044ff0e142cfe5cef8f57d0d1557ac4d1bb253.